### PR TITLE
Update Authorization to v2022-04-01

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Adds support for the latest non-preview version of Authorization

**Special notes for your reviewer**:

Partial resolution of #3294

**Prerequisites**

- [ ] #3501 

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/azure-arm.yaml [hunk 1]
```diff
@@ -515,6 +515,11 @@ status:
       basePath: 'securityinsights/resource-manager/common',
       namespace: 'Microsoft.SecurityInsights'
     },
+    {
```
Comment (matthchr) [inline v2/azure-arm.yaml:518]: Does this change need to be rebased onto master now that the submodule update has been done?

**Review by matthchr — COMMENTED** [review]
Needs a rebase but I think looks good other than that (can't quite tell because of combination of changes due to submodule update).

File: v2/azure-arm.yaml (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/azure-arm.yaml]: Yes. I've rebased and force pushed.

File: v2/samples/authorization/v1api20220401/v1api20220401_roleassignment.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    group: resources.azure.com
+    kind: ResourceGroup
+  # This is the Principal ID of the AAD identity to which the role will be assigned
+  principalIdFromConfig:
+    name: identity-settings
+    key: principalId
+  roleDefinitionReference:
+    # This ARM ID represents "Contributor" - you can read about other built in roles here: https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
+    armId: /subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c
+
```
Comment (matthchr) [inline v2/samples/authorization/v1api20220401/v1api20220401_roleassignment.yaml:19]: minor: Extra newline here

**Review by matthchr — APPROVED** [review]

**Review by super-harsh — APPROVED** [review]

**Comment by matthchr** [comment]
Looks like this needs to re-run `task` (or just the generator) locally after the merge of REST capabilities on resources.

**Pull request merged**